scripts/utils/generate_ts.sh

```diff
@@ -21,82 +21,101 @@ mkdir -p cache || die
 cp scripts/utils/generate_strings.py cache || die
 print G "done."
 
-printn Y "Generating strings... "
-python cache/generate_strings.py translations/strings.yaml -o translations/generated
-print G "done."
+for project in src/apps/*; do
+  project=$(basename $project)
+
+  mkdir -p translations/generated/$project || die
 
-printn Y "Generating a dummy PRO file... "
-mkdir -p translations/generated || die
-cat > translations/generated/dummy_ts.pro << EOF
+  printn Y "$project - Generating strings... "
+  python3 cache/generate_strings.py src/apps/$project/translations/strings.yaml -o translations/generated/$project
+  print G "done."
+
+  printn Y "$project - Generating a dummy PRO file... "
+  cat > translations/generated/$project/dummy_ts.pro << EOF
 HEADERS += i18nstrings.h
 HEADERS += \$\$files(../../src/shared/*.h, true)
```

Comment:
Path should go 3 levels up, not 2? Also, this is not going to work on branches?

Reply:
yes it does because shared/apps are not "old" enough. We support only 2.13 and 2.14 releases.

Reply:
I was talking about all paths, not just this one. You can check the .ts file generated, it's missing all strings but the ones defined in strings.yaml